ckks/bootstrapping/default_params.go

```diff
@@ -24,674 +22,174 @@ var DefaultParametersDense = []defaultParametersLiteral{N16QP1767H32768H32, N16Q
-	// Residual Q : 285 bits.
+	// Residual Q : []int{60, 45, 45, 45, 45} (285 bits).
+	// SlotsToCoeffs Q: []int{42, 42, 42}.
+	// EvalMod Q: []int{60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60}.
+	// CoeffsToSlots Q: []int{58, 58, 58, 58}.
 	// Precision : 32.1 bits for 2^{15} slots.
 	// Failure : 2^{-138.7} for 2^{15} slots.
 	N16QP1547H192H32 = defaultParametersLiteral{
 		ckks.ParametersLiteral{
-			LogN:  16,
-			Sigma: rlwe.DefaultSigma,
-			H:     192,
-			Q: []uint64{
-				0x10000000006e0001, // 60 Q0
-				0x2000000a0001,     // 45
-				0x2000000e0001,     // 45
-				0x1fffffc20001,     // 45
-				0x200000440001,     // 45
-				0x200000500001,     // 45
-				0x3ffffe80001,      // 42 StC
-				0x3ffffd20001,      // 42 StC
-				0x3ffffca0001,      // 42 StC
-				0xffffffffffc0001,  // 60 ArcSine
-				0xfffffffff240001,  // 60 ArcSine
-				0x1000000000f00001, // 60 ArcSine
-				0xfffffffff840001,  // 60 Double angle
-				0x1000000000860001, // 60 Double angle
-				0xfffffffff6a0001,  // 60 Sine
-				0x1000000000980001, // 60 Sine
-				0xfffffffff5a0001,  // 60 Sine
-				0x1000000000b00001, // 60 Sine
-				0x1000000000ce0001, // 60 Sine
-				0xfffffffff2a0001,  // 60 Sine
-				0x400000000360001,  // 58 CtS
-				0x3ffffffffbe0001,  // 58 CtS
-				0x400000000660001,  // 58 CtS
-				0x4000000008a0001,  // 58 CtS
-			},
-			P: []uint64{
-				0x1fffffffffe00001, // Pi 61
-				0x1fffffffffc80001, // Pi 61
-				0x1fffffffffb40001, // Pi 61
-				0x1fffffffff500001, // Pi 61
-			},
-			LogSlots:     15,
-			DefaultScale: 1 << 45,
-		},
-
-		Parameters{
-			EphemeralSecretWeight: 32,
-			SlotsToCoeffsParameters: advanced.EncodingMatrixLiteral{
-				LinearTransformType: advanced.SlotsToCoeffs,
-				RepackImag2Real:     true,
-				LevelStart:          8,
-				BSGSRatio:           2.0,
-				BitReversed:         false,
-				ScalingFactor: [][]float64{
-					{0x3ffffe80001},
-					{0x3ffffd20001},
-					{0x3ffffca0001},
-				},
-			},
-			EvalModParameters: advanced.EvalModLiteral{
-				Q:             0x10000000006e0001,
-				LevelStart:    19,
-				SineType:      advanced.Cos1,
-				MessageRatio:  4.0,
-				K:             16,
-				SineDeg:       30,
-				DoubleAngle:   3,
-				ArcSineDeg:    7,
-				ScalingFactor: 1 << 60,
-			},
-			CoeffsToSlotsParameters: advanced.EncodingMatrixLiteral{
-				LinearTransformType: advanced.CoeffsToSlots,
-				RepackImag2Real:     true,
-				LevelStart:          23,
-				BSGSRatio:           2.0,
-				BitReversed:         false,
-				ScalingFactor: [][]float64{
-					{0x400000000360001},
-					{0x3ffffffffbe0001},
-					{0x400000000660001},
-					{0x4000000008a0001},
-				},
-			},
+			LogN:     16,
+			LogQ:     []int{60, 45, 45, 45, 45, 45},
+			LogP:     []int{61, 61, 61, 61},
+			H:        192,
+			LogScale: 45,
+		},
+		ParametersLiteral{
+			S2CLogScale:     [][]int{{42}, {42}, {42}},
+			C2SLogScale:     [][]int{{58}, {58}, {58}, {58}},
+			LogMessageRatio: 2,
+			ArcSineDeg:      7,
+			EvalModLogScale: 60,
```

Comment:
These `LogScale`s probably have to be bigger than that in `LogScale`? If yes, I think a "constructor" should be use and this invariant should be explicitly checked.

Reply:
These are bootstrapping parameters (homomorphic encoding and decoding), which are independent of the scaling factor. They provide verified performances that are detailed in the godoc. Advanced users can specify different values to customize the bootstrapping circuit.

Reply:
This makes sense. Thank you!